### DISCUSSION TITLE
How to properly type hint IO[bytest]

### DISCUSSION THREAD

**Discussion by randolf-scholz** [python/typing#1293] [Q&A]
I have the following code snippet

```python
from pathlib import Path
from typing import IO
from io import IOBase

def read_file(file: str | Path | IO[bytes]):
    ...
    file.seek(0, 0) if isinstance(file, IOBase) else None
```

And mypy complains:

```
error: Subclass of "IO[bytes]" and "IOBase" cannot exist: would have incompatible method signatures  [unreachable]
```

I thought one was supposed to use `typing.IO` for type hinting here?

**Comment by srittau** [comment]
`IO` classes are weird. I would recommend to use `hasattr(file, "seek")` instead of `isinstance`, although all `IO` classes are guaranteed to have a `seek()` method.
